src/main/webapp/app/exercises/programming/shared/code-editor/service/code-editor-domain-dependent-endpoint.service.ts

```diff
@@ -22,16 +22,18 @@ export abstract class DomainDependentEndpointService extends DomainDependentServ
      */
     setDomain(domain: DomainChange) {
         super.setDomain(domain);
-        const [domainType, domainValue] = this.domain;
+        this.restResourceUrl = this.calculateRestResourceURL(domain);
+    }
+
+    calculateRestResourceURL(domain: DomainChange): string | null {
+        const [domainType, domainValue] = domain;
         switch (domainType) {
             case DomainType.PARTICIPATION:
-                this.restResourceUrl = `${this.restResourceUrlBase}/repository/${domainValue.id}`;
-                break;
+                return `${this.restResourceUrlBase}/repository/${domainValue.id}`;
             case DomainType.TEST_REPOSITORY:
-                this.restResourceUrl = `${this.restResourceUrlBase}/test-repository/${domainValue.id}`;
-                break;
+                return `${this.restResourceUrlBase}/test-repository/${domainValue.id}`;
             default:
-                this.restResourceUrl = null;
+                return null;
```

Comment:
Maybe send an issue to sentry in that case?
Or do you expect this default case to be executed normally?

Reply:
I just moved the old code into a new method here, so I kept it. The enum only has two values so this should actually never happen: 
![grafik](https://user-images.githubusercontent.com/26540346/182389141-13f6168a-69e5-4549-abe1-f49d05994b13.png)

Reply:
If it should not happen it sounds like a quick ``captureException`` call is what we want (isn't Sentry for all the "this should not happen" stuff? :D)